metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1620,6 +1627,24 @@ class MetalsLanguageServer(
     results
   }
 
+  /** Requesting semantic tokens for a whole file in order to highlight */
+  @JsonRequest("textDocument/semanticTokens/full")
+  def semanticTokensFull(
+      params: SemanticTokensParams
+  ): CompletableFuture[SemanticTokens] = {
+    scribe.info("Debug: MetalsLanguageServer.semanticHighlighting: Start")
+
+    CancelTokens.future { token => 
+      compilers.semanticTokens(
+        params
+        ,SemanticTokenCapability.TokenTypes
```

Comment:
We don't need to send it to the compiler if it's not configurable. We can put the object in the same module as the compiler.

Reply:
I want to pass the current capability with  PresentationCompilerConfig. But it will take a bit time so I suspend to apply.